lib/utils.bash

```diff
@@ -806,3 +806,20 @@ with_shim_executable() {
 
   return 126
 }
+
+substitute() {
+  # Use Bash substituion rather than sed as it will handle escaping of all
+  # strings for us.
+  local input=$1
+  local find_str=$2
+  local replace=$3
+  printf "%s" "${input//"$find_str"/"$replace"}"
+}
+
+remove_path_from_path() {
+  # A helper function for removing an arbitrary path from the PATH variable.
+  # Output is a new string suitable for assignment to PATH
+  local PATH=$1
+  local path=$2
+  substitute "$PATH" "$path" "" | sed -e "s|::|:|g"
+}
```

Comment:
I noticed we did this in a couple places so I decided to extract this logic out into it's own function.